### PR TITLE
Specs and standards

### PR DESCRIPTION
**Pull request by jrostand**
I'm not sure how warm a welcome this will get, but I was on a caffeine binge and decided to do some housekeeping around the project, primarily the specs. Here are the highlights:
- Changed a couple of method names to better reflect their "dangerous"/data-changing natures (most notably `Block#update!` and `Manager#run!`)
- Renamed half of the Blocks (Cpu -> CPU, Hdd -> HDD, Ip -> IP, WhoAmI -> Whoami), partly for vanity and partly for regularisation
- Changed the Rhythmbox option names, dropping `show_` in all cases
- Added an IPv6 option to the IP Block
- Brought specs into line with [BetterSpecs](http://betterspecs.org/) (more or less)
- Removed a fair few specs that weren't actually testing gem functionality
- Moved a couple of the test mock objects to `spec/mocks/`, with references as appropriate, and got rid of the rest in favor of partial doubles
- Generally brought things into line with a styleguide, more for consistency than anything else

tl;dr if you're into that sort of thing: I basically changed a lot of stuff. There are few untouched files. I can only apologize and blame insomnia.

I genuinely don't know whether or not I've seriously broken anything (apart from backwards-compatibility). I know all the specs pass, but that says very little. I also tested as much as I could with the examples, but I don't use i3, so that should probably get a thorough going-over.

Well, there you have it. giant pull request. Quite a way to start public involvement in a project, isn't it? I look forward to the comments below.

### CODE REVIEW

**Comment by OkayDave** [comment]
Thanks a bunch for the PR! I'll have a read through and play with it. I have a couple of comments before I start:

I know very little about rspec (I use minitest in my day job) so any and all improvements there are welcome :+1: 

I'm not opposed to breaking backwards compatibility per se, I'd just rather wait a bit longer than day three :). That said, I'd much rather the base code was 'right' and then have compatibility maintained with aliases.

**Comment by OkayDave** [comment]
This is really good; thank you for this @jrostand. Your changes to the system calls a great and your other changes make a lot of sense. I3 still works :+1:. 

The only thing I dislike is changing the  `@output` var to `@data`, though I get that `@output` perhaps isn't the best name for it, either. 

Before I merge it in to master I'm going to add a bit of compatibility code. I'll then add deprecation warnings in a later release.

**Comment by jrostand** [comment]
I'm very pleased to hear that you like what I've done. Fair play on `@output`/`@data` - as you point out, I was trying to make its name better fit its function, and it's possible that neither is the best moniker. Thanks for being so accommodating of my sweeping reforms.

**Comment by OkayDave** [comment]
Closing this PR. @jrostand I've incorporated your commits into #8
